### PR TITLE
[REVIEW] Fix DGA Detection Notebook and DetectorDataset Memory Errors 

### PR DESCRIPTION
**Pull request by bsuryadevara**
Addressed multiple issues
- Fixed DGA Detection notebook exception(unable to pull public dataset).
- Removed DetectorDataset to overcome memory errors if batch size is set to a small number for model training.
- Introduced `DataLoader`, `Dataset` and `DGADataset`(custom dataset which inherits `Dataset`) wrapper classes to support model training even with small batches.
- Created test scripts for `DataLoader` and `DGADataset`.
- Updated `run_dga_training.py` and `dga_detection.py` to align with the above changes.

### CODE REVIEW

**Comment by bsuryadevara** [comment]
rerun tests

File: python/clx/analytics/dga_detector.py [hunk 1]
```diff
@@ -63,26 +63,25 @@ def save_model(self, file_path):
         }
         super()._save_model(checkpoint, file_path)
 
-    def train_model(self, detector_dataset):
+    def train_model(self, dataloader):
```
Comment (efajardo-nv) [inline python/clx/analytics/dga_detector.py:66]: what about hiding the use of dataloader and dataset? might be easier for someone to just pass cudf dataframe directly along with batch_size and avoid the preprocessing. we can do that under the hood.

Comment (bsuryadevara) [reply python/clx/analytics/dga_detector.py:66]: @efajardo-nv we can pass the dataframe directly, but `train_model` function is called for every epoch and preprocessing happens repeatedly. Do you think adding a separate function which reads cudf dataframe and returns dataloader is good idea?

Comment (efajardo-nv) [reply python/clx/analytics/dga_detector.py:66]: can we have `train_model` perform the epochs internally as well?

Comment (bsuryadevara) [reply python/clx/analytics/dga_detector.py:66]: I can make changes within the train model function to run multiple epochs.

Comment (bsuryadevara) [reply python/clx/analytics/dga_detector.py]: Updated with the suggested changes. But currently `train_model` test is failing with at cuML function `train_test_split`. Thank you!

**Comment by bsuryadevara** [comment]
rerun tests

**Comment by efajardo-nv** [comment]
rerun tests

**Comment by bsuryadevara** [comment]
rerun tests

**Comment by bsuryadevara** [comment]
rerun tests

**Comment by efajardo-nv** [comment]
rerun tests

**Comment by efajardo-nv** [comment]
rerun tests

**Comment by efajardo-nv** [comment]
rerun tests

**Review by efajardo-nv — APPROVED** [review]

**Review by BartleyR — APPROVED** [review]

**Pull request merged**